File: typeface-views/src/main/java/com/sorcererxw/typefaceviews/preferences/TypefacePreferenceCategory.java
package com.sorcererxw.typefaceviews.preferences;

import android.content.Context;
import android.graphics.Typeface;
import android.preference.PreferenceCategory;
import android.util.AttributeSet;
import android.view.View;
import android.widget.TextView;

import com.sorcererxw.typefaceviews.TypefaceViews;

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/10/13
 */

public class TypefacePreferenceCategory extends PreferenceCategory {

    public TypefacePreferenceCategory(Context context, AttributeSet attrs, int defStyleAttr) {
        super(context, attrs, defStyleAttr);
    }

    public TypefacePreferenceCategory(Context context, AttributeSet attrs) {
        super(context, attrs);
    }

    public TypefacePreferenceCategory(Context context) {
        super(context);
    }

    private View mView;

    private TextView mTitleTextView;

    @Override
    protected void onBindView(View view) {
        super.onBindView(view);
        mView = view;
        mTitleTextView = (TextView) mView.findViewById(android.R.id.title);
        mTitleTextView.setTextColor(ColorStateLists.TitleColorStateList(getContext()));
        if (getTitle() != null && getTitle().length() > 0) {
            setTitleTypeface(TypefaceViews.getMediumTypeface());
        }
    }

    public View getView() {
        return mView;
    }

    public void setTitleTypeface(Typeface typeface) {
        if (mTitleTextView == null) {
            return;
        }
        mTitleTextView.setTypeface(typeface);
    }
}

File: typeface-views/src/main/java/com/sorcererxw/typefaceviews/preferences/TypefacePreferenceCategory.kt
package com.sorcererxw.typefaceviews.preferences

import android.content.Context
import android.graphics.Typeface
import android.preference.PreferenceCategory
import android.util.AttributeSet
import android.view.View
import android.widget.TextView

import com.sorcererxw.typefaceviews.TypefaceViews

/**
 * @description:
 * @author: Sorcerer
 * @date: 2016/10/13
 */

class TypefacePreferenceCategory : PreferenceCategory {

    var view: View? = null
        private set

    private var mTitleTextView: TextView? = null

    constructor(context: Context, attrs: AttributeSet, defStyleAttr: Int) : super(context, attrs,
            defStyleAttr) {
    }

    constructor(context: Context, attrs: AttributeSet) : super(context, attrs) {}

    constructor(context: Context) : super(context) {}

    override fun onBindView(view: View) {
        super.onBindView(view)
        this.view = view
        mTitleTextView = this.view!!.findViewById<View>(android.R.id.title) as TextView
        mTitleTextView!!.setTextColor(ColorStateLists.TitleColorStateList(context))
        if (title != null && title.length > 0) {
            setTitleTypeface(TypefaceViews.mediumTypeface)
        }
    }

    fun setTitleTypeface(typeface: Typeface?) {
        if (mTitleTextView == null) {
            return
        }
        mTitleTextView!!.typeface = typeface
    }
}
